ARC: Correction of uImage target build

xxkent opened the pull request:
"uImage" target now have proper dependency on "loader". "make uImage" works properly.

Add a GitHub CI test builds to ci.yml for automated testing of uImage building.

xxkent committed:
ARCv3: Add missing dependency for uImage

Using clean build of uImage target by "make uImage" resulted
in an error "'arch/arc/boot/loader': No such file".
This was because of "loader" is not dependancy for "uImage".

With this fix "make clean"+"make uImage" generate
"loader" and "uImage" binaries in "arch/arc/boot/".

abrodkin reviewed (changes requested):
Please separate CI changes (.github/workflows/ci.yml) from the actual kernel fixes. That will help us to rebase it and upstream when the time comes.

xxkent committed:
ARC: Github CI: Add uImage binaries building

Add a GitHub CI test builds to ci.yml for automated testing of
uImage building. This adds uImage artifacts files in addition to
vmlinux files.

abrodkin reviewed (changes requested):
Please reword the title to something more meaningful like: "ARCv3: Add missing dependency for uImage" and then in the commit message explain the problem (forget about initial GitHub issue as it won't be visible in a pure git tree, for example when somebody will look at his local source tree) and why that's the right solution.
